main.ts

```diff
@@ -176,6 +184,71 @@ const openGistEditorCallback =
     }
   };
 
+const deleteGistEditorCallback =
+  (opts: DeleteGistEditorCallbackParams) => async () => {
+    const { app, plugin } = opts;
+
+    const view = app.workspace.getActiveViewOfType(MarkdownView);
+
+    if (!view) {
+      return new Notice('No active file');
+    }
+
+    const editor = view.editor;
+    const originalContent = editor.getValue();
+
+    const existingSharedGists = getSharedGistsForFile(originalContent);
+
+    if (existingSharedGists.length === 0) {
+      const { enableUpdatingGistsAfterCreation } =
+        await getLatestSettings(plugin);
+
+      if (!enableUpdatingGistsAfterCreation) {
+        return new Notice(
+          "You need to enable 'Update gists after creation' in Settings to use this command.",
+        );
+      } else {
+        return new Notice('There are no gists associated with this note.');
+      }
+    }
+
+    if (existingSharedGists.length > 1) {
```

Comment:
For now, I'm following the existing very repetitive pattern for dealing with the two cases: one gist and more than one gist. But I should abstract this away at some point for less repetition.